.github/workflows/ci.yml

```diff
@@ -44,9 +44,9 @@ jobs:
     - run: yarn i18n:checks
     - run: yarn dogfood-lhci
 
-    # buildtracker runs `git merge-base HEAD origin/master` which needs more history than depth=1. https://github.com/paularmstrong/build-tracker/issues/106
-    - name: Deepen git fetch (for buildtracker)
-      run: git fetch --deepen=100
+    # buildtracker needs history and a common merge commit.
```

Comment:
Could drop the entire script if we just skip this step if the current branch isn't master? so it only runs on commits merged to master.

Reply:
this script is mostly necessary for PRs.

though on master we need we need just 1 command... the `git fetch --deepen`.
the bash script early exists (line 26) in this case.

Reply:
Why do we want it tracking PR? It doesn't ever fail, right?

Reply:
we want master for timeseries historical data 

we want prs for budgets. once bt is running successfully i can add [bundle size budgets](https://buildtracker.dev/docs/budgets#type), which is the last remaining item to complete the move from the old `bundlesize` module to this. 
buildtracker CLI won't have a non-zero exit code, even with failing budgets.. but i'll likely [post to the status API](https://github.com/paularmstrong/build-tracker/issues/199) to express the failure state

Reply:
Is there a way to check against a budget w/o uploading the data for the PR revision (and thus, no need to do this complex git stuff)?

Reply:
If build tracker UI can't filter to just master commits, I'm hesitant to mix the two data sources.

Reply:
> If build tracker UI can't filter to just master commits, I'm hesitant to mix the two data sources.


totally. the UI filters to just master/default branch by default. so they're not mixed.


> Is there a way to check against a budget w/o uploading the data for the PR revision (and thus, no need to do this complex git stuff)?

nope. while you can set 3 [types of budgets](https://buildtracker.dev/docs/budgets#type) (SIZE, DELTA, PERCENT_DELTA), the type we want is DELTA, which needs to know the baseline size, and only the server knows that. so in BT, the budgets are configured against the server instead of the client who's doing the build.

i asked the buildtracker maintainer about uploading non-master stuff and he said at twitter they upload all branches. BT may at some point allow switching the UI to look at a diff branch but its not implemented yet.